.github/workflows/linux.yml

```diff
@@ -16,476 +16,490 @@ concurrency:
+        working-directory: ${{ env.OV_INSTALL_DIR }}
+        
+      - name: Build Samples (Release)
+        if: ${{ 'Release' == matrix.build-type }}
         run: |
-          echo 'Acquire::Retries "10";' | sudo tee -a /etc/apt/apt.conf.d/80-retries > /dev/null
-          echo 'APT::Get::Assume-Yes "true";' | sudo tee -a /etc/apt/apt.conf.d/81-assume-yes > /dev/null
-          echo 'APT::Get::Fix-Broken "true";' | sudo tee -a /etc/apt/apt.conf.d/82-fix-broken > /dev/null
-          echo 'APT::Get::no-install-recommends "true";' | sudo tee -a /etc/apt/apt.conf.d/83-no-recommends > /dev/null
-
-      - name: Install build dependencies
+          chmod +x ${{ env.OV_INSTALL_DIR }}/samples/cpp/build_samples.sh
+          ${{ env.OV_INSTALL_DIR }}/samples/cpp/build_samples.sh -i ${{ env.INSTALL_DIR }}
+  
+      - name: Build Samples (${{ matrix.build-type }})
+        if: ${{ 'Release' != matrix.build-type }}
         run: |
-          sudo ${OV_INSTALL_DIR}/install_dependencies/install_openvino_dependencies.sh
-          sudo apt-get install ccache
+          source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
+          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${{ env.OV_INSTALL_DIR }}/samples/cpp/ -B ${{ env.BUILD_DIR }}
+          cmake --build ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --parallel $(nproc)
+          cmake --install ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --component samples_bin --prefix ${{ env.INSTALL_DIR }}
+        
+      - name: Pack Artifacts
+        run: tar -cvf - * | pigz > ${{ env.INSTALL_DIR }}/${{ env.OV_SAMPLES_NAME }}
+        working-directory: ${{ env.INSTALL_DIR }}
 
-      - name: Setup ccache
-        uses: actions/cache@v4
+      - name: Upload Samples Build Package
+        if: ${{ always() }}
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
         with:
-          # Should save cache only if run in the master branch of the base repo
-          # github.ref_name is 'ref/PR_#' in case of the PR, and 'branch_name' when executed on push
-          save-always: ${{ github.ref_name == 'master' && 'true' || 'false'  }}
-          path: ${{ env.CCACHE_DIR }}
-          key: ${{ runner.os }}-${{ runner.arch }}-ccache-genai-release-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-${{ runner.arch }}-ccache-genai-release
+          name: ov_samples_${{ matrix.build-type }}
+          path: ${{ env.INSTALL_DIR }}/*.tar.gz
+          if-no-files-found: 'error'
+          
+  genai_tests_archive:
+    name: Test ${{ matrix.test.name }} (archive/${{ matrix.build-type }})
+    needs: [ openvino_download, genai_build_cmake ]
+    strategy:
+      fail-fast: false
+      matrix:
+        build-type: [Release]
+        test:
+          - name: 'Chat Generate Api'
+            cmd: 'tests/python_tests/test_chat_generate_api.py -k "not Qwen2-0.5B-Instruct"'
```

Comment:
you have dropped testing of other scenarios:
- https://github.com/openvinotoolkit/openvino.genai/blob/master/tests/python_tests/test_cache_optimizations.py
- https://github.com/openvinotoolkit/openvino.genai/blob/master/tests/python_tests/test_generate_api.py
- https://github.com/openvinotoolkit/openvino.genai/blob/master/tests/python_tests/test_llm_pipeline_static.py
- https://github.com/openvinotoolkit/openvino.genai/blob/master/tests/python_tests/test_preemption.py
- https://github.com/openvinotoolkit/openvino.genai/blob/master/tests/python_tests/test_sampling.py

they are tested here:
https://github.com/openvinotoolkit/openvino.genai/blob/ee91fcfa1b43c7cd42be58688832318fe6e8b0af/.github/workflows/linux.yml#L266-L271